### ISSUE TITLE
Issue with engine eval tags

### ISSUE THREAD

**Issue by ExtremeCoolDude** [mliebelt/PgnViewerJS#203]
`{ [%eval 1.20]} `Comments like this produce an error and fails the pgn to parse at all.  There are some highlight squares, draw arrows operation inside { } but an option to not process anything whatever is inside { } would be good. 

For eg :  on https://mliebelt.github.io/PgnViewerJS/config/config.html, This will produce an error : 

```
[Event "?"]
[Site "?"]
[Date "????.??.??"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "*"]
[ECO "E14"]

1. d4 d5 2. Nf3  { [%eval 1.20]} 
```

This will not produce an error : 
```

[Event "?"]
[Site "?"]
[Date "????.??.??"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "*"]
[ECO "E14"]

1. d4 d5 2. Nf3 
```

**Comment by mliebelt** [comment]
Thank you for raising that issue. I will address it at the weekend. My plan is (in that order):

1. Add to the grammar the possibility to parse anything that starts with `[%` and ignore it then.
2. Add (later) the possibility to use the eval values.
3. Use that version in `pgn-reader` and increment the version.
4. Use that new version in `pgn-viewer`.

I am currently under way in migrating to the latest version of `chessground`, that takes some time. And  only after that is finished I will address steps 3 upwards.

**Comment by ExtremeCoolDude** [comment]
Thanks a lot. I'd very much appreciate if you provided option (or advice if its not possible on code)  to ignore everything inside comments tag at all. For instance this produces error : 

```
{1. Hello Bf8 f8-a3 (hello to pawn on e6-f7-g6-h7) 2. I like Bf4 (or Bh3) 3. Probably not h3 (because why not me & me vs the world). White wants the center
so black plays Sicilian. Another thing is socrates was muscular but light is grandmaster than feather bishop (Bb7 is green). Remember "limitation" is the mother of inventions in strategic pawn sacrific, especially when you have space advantage.} 
```

I cannot reproduce error on your config website with above comment. But I'm using the pgnViewerJS inside Anki software with https://github.com/TowelSniffer/Anki-Chess-2.0 and this is the error screenshot : 

![image](https://user-images.githubusercontent.com/50034073/112531487-bc30e600-8dcf-11eb-9d1e-56042d20100e.png)

**Comment by mliebelt** [comment]
So I implemented the grammar already, see the tickets [#40](https://github.com/mliebelt/pgn-parser/issues/40) and [#41](https://github.com/mliebelt/pgn-parser/issues/41) for details.

About that Anki-Chess .. (I was not aware of that): It seems to use an old version of PgnViewerJS, but in the README, it states that it uses as well pgn-parser (which is quite new). So I am not sure how that works together. It must be an old version, because I switched to Chessground some time ago from chessboardjs (which was nice). Or the README (and animated GIF) are old.

**Comment by ExtremeCoolDude** [comment]
Thanks a lot! The %eval tag is working! However, I'm still getting pegsyntax error on the above comment, I found the culprit to be `&`, when I remove this character, the pgn parses correctly! 

 The gif is old as it shows chesstempo's viewer (the blue one on the gif) which was kind of buggy. It was switched to PGNViewerJS a month ago I think (via Browserify)

Anki cards have front and back, first the user input PGN is read from front of card and parsed using pgn-parser to move array and fed to chessboardjs one move at a time. This is done so that the flash cards are interactive and allow the users to memorize opening or play  tactics move by move.  And the back of the card uses PgnviewerJS (slightly customized on my PC to have board background) which allows to view all the sequence of moves: 

<a href="https://gifyu.com/image/Ybr0"><img src="https://s4.gifyu.com/images/peekfin.gif" alt="peekfin.gif" border="0" /></a>

**Comment by mliebelt** [comment]
To your above error: the string 

```
{1. Hello Bf8 f8-a3 (hello to pawn on e6-f7-g6-h7) 2. I like Bf4 (or Bh3) 3. Probably not h3 (because why not me & me vs the world). White wants the center
so black plays Sicilian. Another thing is socrates was muscular but light is grandmaster than feather bishop (Bb7 is green). Remember "limitation" is the mother of inventions in strategic pawn sacrific, especially when you have space advantage.}
```

is not really a pgn string. When I add after the string " 1. e4" (to have a move), it works. I will check if that is a legal input for a board.

**Comment by ExtremeCoolDude** [comment]
I meant it after the moves xD for eg : 

```
[Event "?"]
[Site "?"]
[Date "????.??.??"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "*"]
[ECO "E14"]

1. d4 d5 2. Nf3 { Here black & white look good } 
```

gives out ` Uncaught peg$SyntaxError` where & is the culprit. Thanks a lot again for the wonderful parser!

**Comment by mliebelt** [comment]
Here is the test in my development environment in debug mode:

**Comment by ExtremeCoolDude** [comment]
Thank you so much, Perhaps there's some issue in getting in pgn via encodeURI or something, I'll have to  look it up, and a new ticket if its related to pgn-parser. Thanks again!

**Comment by ExtremeCoolDude** [comment]
My bad, the issue is indeed unrelated to pgn-parser, since there's no way to get the data from within Anki to external scripts, its done by getting the value via URL address, which results in truncated output after `&` and malformed input into pgn-parser : 
![image](https://user-images.githubusercontent.com/50034073/112717273-a6c4d480-8f13-11eb-8a37-2b5a40bf4159.png)

**Comment by mliebelt** [comment]
@ExtremeCoolDude So you have URL-encode it. I don't know which technology you are using, but hopefully there is a way to use a library call to do that.

**Comment by ExtremeCoolDude** [comment]
its just plain vanilla js, it is encoded 
```
<script>
var FEN = encodeURI(`{{text:FEN}}`);
var Title = encodeURI(`{{text:Title}}`);
var Comments = encodeURI(`{{text:Comments}}`);
var PGN = encodeURI(`{{text:PGN}}`);

document.getElementById("Board").src =`_board.html?PGN=`+PGN+`&FEN=`+FEN+`&Comments=`+Comments+`&Title=`+Title+``;
</script>
```

I've opened a issue in @TowelSniffer's repo, not sure what's causing it.
